tests/e2e/test_spyre_cb.py

```diff
@@ -324,7 +327,298 @@ def get_params_test_blocks_borders_misaligned_prompts():
+        },
+        {
+            # Decode sequences 0 and 1
+            "step": 9,
+            "tkv": 71,
+            "waiting": [],
+            "running": ["1", "0"],
+            "request_outputs": ["1", "0"]
+        },
+        {
+            # Sequence 0 finishes at step 58
+            # (start step + 2 prefills + 56 decodes - 1) = 1 + 2 + 56 - 1 = 58
+            "step": 58,
+            "tkv": 120,
+            "waiting": [],
+            "running": ["1"],
+            "request_outputs": ["1", "0"],
+            "finished_requests": ["0"]
+        },
+        {
+            # Decode sequence 1
+            "step": 59,
+            "tkv": 121,
+            "waiting": [],
+            "running": ["1"],
+            "request_outputs": ["1"],
+        },
+        {
+            # Sequence 1 finishes at step 74
+            # (start step + 1 prefill + 66 decodes - 1) = 8 + 1 + 66 - 1 = 74
+            "step": 74,
+            "tkv": 136,
+            "waiting": [],
+            "running": [],
+            "request_outputs": ["1"],
+            "finished_requests": ["1"]
+        },
+        {
+            # Tkv should be cleared one step later
+            "step": 75,
+            "tkv": 0,
+            "waiting": [],
+            "running": [],
+            "request_outputs": []
+        },
+    ]
+
+    return (seqs_max_tokens, prompts_lengths, steps_add_reqs, checked_steps,
+            max_model_len)
+
+
+def get_params_test_scheduler_constraints_max_prompt_len():
+    """ Scenario where the request goes beyond max_model_len """
+
+    seqs_max_tokens = [67, 57, 80]
+    prompts_lengths = [70, 49, 41]
+    steps_add_reqs = [0, 0, 0]
+    max_model_len = 256
+
+    checked_steps = [
+        {
+            "step": 0,
+            "tkv": 0,
+            "waiting": ["0", "1", "2"],
+            "running": [],
+            "request_outputs": []
+        },
+        {
+            # Prefill sequence 0
+            "step": 1,
+            "tkv": 128,
+            "waiting": ["1", "2"],
+            "running": ["0"],
+            "request_outputs": ["0"]
+        },
+        {
+            # Prefill sequence 1
+            "step": 2,
+            "tkv": 128,
+            "waiting": ["2"],
+            "running": ["1", "0"],
+            "request_outputs": ["1"]
+        },
+        {
+            # Decode sequences 0 and 1
+            "step": 3,
+            "tkv": 129,
+            "waiting": ["2"],
+            "running": ["1", "0"],
+            "request_outputs": ["1", "0"]
+        },
+        {
+            # Sequence 1 finishes at step 58
+            # (start step + 1 prefills + 56 decodes - 1) = 2 + 1 + 56 - 1 = 58
+            "step": 58,
+            "tkv": 184,
+            "waiting": ["2"],
+            "running": ["0"],
+            "request_outputs": ["1", "0"],
+            "finished_requests": ["1"]
+        },
+        {
+            # Decode sequence 0
+            # Cannot decode sequence 2: 185 + 80 = 265 > 256
```

Comment:
should this say `prefill` instead?
```suggestion
            # Cannot prefill sequence 2: 185 + 80 = 265 > 256
```

Reply:
good catch!